### ISSUE TITLE
nil pointer dereference when friendbot not set

### ISSUE THREAD

**Issue by bartekn** [stellar/go#642]
```
*errors.Error: runtime error: invalid memory address or nil pointer dereference
  File "/usr/lib/go-1.9.7/src/runtime/panic.go", line 63, in panicmem
  File "/usr/lib/go-1.9.7/src/runtime/signal_unix.go", line 367, in sigpanic
  File "/usr/lib/go-1.9.7/src/net/url/url.go", line 724, in String
  File "/var/lib/jenkins/jobs/stellar-horizon-public/workspace/.go/src/github.com/stellar/go/services/horizon/internal/init_web.go", line 150, in func6
...
```

**Comment by bartekn** [comment]
@rudle you may want to check it out.

**Comment by rudle** [comment]
Thanks @bartekn, I have a quickfix in progress. Can you share with me the Jenkins job where this failed? I'd like to verify my fix is a good one.

**Comment by bartekn** [comment]
This is not a Jenkins job, source file location is hardcoded is the Go binary and this one was built in Jenkins. But the bug is [here](https://github.com/stellar/go/blob/b20aa6670f1220a832403ae5f2b3f1e3b0cc83b2/services/horizon/internal/init_web.go#L150). We should probably not expose `/friendbot` at all when `app.config.FriendbotURL` is `nil`.

**Comment by rudle** [comment]
I agree, and I've prepared a change here https://github.com/stellar/go/compare/master...rudle:friendbot-npe

I was hoping to find a failing test case so that I could verify my fix was sound. I will write my own regression test case.

**Comment by bartekn** [comment]
Yeah, this looks good. Please create a PR when you have a moment and I'll merge it.

**Comment by rudle** [comment]
https://github.com/stellar/go/pull/646

**Comment by bartekn** [comment]
Closed in #646.
